plugins/modules/external_account_info.py

```diff
@@ -0,0 +1,213 @@
+    required: no
+    choices:
+      - AWS_ACCESS_KEY_AUTH
+      - AWS_IAM_ROLES_AUTH
+      - ALTUS_ACCESS_KEY_AUTH
+      - ADLS_AD_SVC_PRINC_AUTH
+      - BASIC_AUTH
+"""
+
+EXAMPLES = r"""
+---
+- name: Get all external accounts in Cloudera Manager
+  cloudera.cluster.external_account_info:
+    host: example.cloudera.com
+    port: "7180"
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+
+- name: Get all external accounts with type "AWS_ACCESS_KEY_AUTH"
+  cloudera.cluster.external_account_info:
+    host: example.cloudera.com
+    port: "7180"
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    type: "AWS_ACCESS_KEY_AUTH"
+
+- name: Get specific external account
+  cloudera.cluster.external_account_info:
+    host: example.cloudera.com
+    port: "7180"
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: "aws_test_key"
+
+"""
+
+RETURN = r"""
+---
+external_account:
+    description:
+        - List of one or more external accounts within the cluster.
+    type: list
+    elements: dict
+    returned: always
+    contains:
+        name:
+            description: Represents the initial name of the account.
+            type: str
+            returned: always
+        displayName:
+            description: A modifiable label to identify this account for user-visible purposes.
+            type: str
+            returned: always
+        createdTime:
+            description: The time of creation for this account.
+            type: str
+            returned: always
+        lastModifiedTime:
+            description: The last modification time for this account.
+            type: str
+            returned: always
+        typeName:
+            description: The Type ID of a supported external account type.
+            type: str
+            returned: always
+        accountConfigs:
+            description: The configuration options for this account.
+            type: list
+            elements: dict
+            returned: always
+"""
+
+
+class ClouderaExternalAccountInfo(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaExternalAccountInfo, self).__init__(module)
+
+        # Set the parameters
+        self.name = self.get_param("name")
+        self.type = self.get_param("type")
+
+        # Initialize the return values
+        self.external_account_info_output = []
+        self.changed = False
+
+        # Execute the logic
+        self.process()
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        api_instance = ExternalAccountsResourceApi(self.api_client)
+        account_types = [
+            "AWS_ACCESS_KEY_AUTH",
+            "AWS_IAM_ROLES_AUTH",
+            "ALTUS_ACCESS_KEY_AUTH",
+            "ADLS_AD_SVC_PRINC_AUTH",
+            "BASIC_AUTH",
+        ]
+        try:
+            if self.name:
+                self.external_account_info_output = [
+                    api_instance.read_account(self.name).to_dict()
```

Comment:
Not sure why this doesn't snake-case the parameters to be Pythonic from the camelCase of Java, but we should endeavor to make this change.